ccl/backupccl: TestFullClusterBackup failed

[(ccl/backupccl).TestFullClusterBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1991572&tab=buildLog) on [master@c9b9c01cd622372cc2c52a8f1d60aa3a5393e785](https://github.com/cockroachdb/cockroach/commits/c9b9c01cd622372cc2c52a8f1d60aa3a5393e785):

```
I200605 16:28:25.568387 444994 kv/kvserver/replica_command.go:413  [n1,s1,r103/1:/Table/5{3/1/2-8}] initiating a split of this range at key /Table/53/1/3 [r105] (manual)
I200605 16:28:25.650619 444801 kv/kvserver/replica_command.go:413  [n1,s1,r103/1:/Table/5{3/1/2-8}] initiating a split of this range at key /Table/53/1/4 [r106] (manual)
I200605 16:28:25.863754 444801 kv/kvserver/replica_command.go:413  [n1,s1,r105/1:/Table/5{3/1/3-8}] initiating a split of this range at key /Table/53/1/4 [r107] (manual)
I200605 16:28:26.099537 444801 kv/kvserver/replica_command.go:413  [n1,s1,r107/1:/Table/5{3/1/4-8}] initiating a split of this range at key /Table/53/1/8 [r108] (manual)
I200605 16:28:26.102876 444995 kv/kvserver/replica_command.go:413  [n1,s1,r107/1:/Table/5{3/1/4-8}] initiating a split of this range at key /Table/53/1/5 [r109] (manual)
I200605 16:28:26.131115 309394 server/status/runtime.go:499  [n1] runtime stats: 6.0 GiB RSS, 406 goroutines, 544 MiB/292 MiB/1010 MiB GO alloc/idle/total, 64 MiB/108 MiB CGO alloc/total, 382.7 CGO/sec, 130.5/12.7 %(u/s)time, 0.2 %gc (1x), 2.2 MiB/2.2 MiB (r/w)net
I200605 16:28:26.352933 444995 kv/kvserver/replica_command.go:413  [n1,s1,r109/1:/Table/5{3/1/5-8}] initiating a split of this range at key /Table/53/1/6 [r110] (manual)
I200605 16:28:26.459331 444801 kv/kvserver/replica_command.go:413  [n1,s1,r109/1:/Table/5{3/1/5-8}] initiating a split of this range at key /Table/53/1/8 [r111] (manual)
I200605 16:28:26.642621 444995 kv/kvserver/replica_command.go:413  [n1,s1,r110/1:/Table/5{3/1/6-8}] initiating a split of this range at key /Table/53/1/7 [r112] (manual)
I200605 16:28:26.732373 444801 kv/kvserver/replica_command.go:413  [n1,s1,r110/1:/Table/5{3/1/6-8}] initiating a split of this range at key /Table/53/1/8 [r113] (manual)
I200605 16:28:26.890147 444801 kv/kvserver/replica_command.go:413  [n1,s1,r112/1:/Table/5{3/1/7-8}] initiating a split of this range at key /Table/53/1/8 [r114] (manual)
I200605 16:28:27.146862 444994 kv/kvserver/replica_command.go:413  [n1,s1,r114/1:/Table/5{3/1/8-8}] initiating a split of this range at key /Table/53/1/9 [r115] (manual)
I200605 16:28:27.377289 444994 kv/kvserver/replica_command.go:413  [n1,s1,r115/1:/Table/5{3/1/9-8}] initiating a split of this range at key /Table/54 [r116] (manual)
I200605 16:28:27.714251 444994 kv/kvserver/replica_command.go:413  [n1,s1,r116/1:/Table/5{4-8}] initiating a split of this range at key /Table/56 [r117] (manual)
I200605 16:28:30.691572 311116 server/status/runtime.go:499  [n1] runtime stats: 6.0 GiB RSS, 394 goroutines, 576 MiB/271 MiB/1010 MiB GO alloc/idle/total, 63 MiB/108 MiB CGO alloc/total, 501.8 CGO/sec, 101.5/6.0 %(u/s)time, 0.0 %gc (0x), 2.3 MiB/2.3 MiB (r/w)net
I200605 16:28:36.131657 309394 server/status/runtime.go:499  [n1] runtime stats: 6.0 GiB RSS, 395 goroutines, 598 MiB/252 MiB/1010 MiB GO alloc/idle/total, 63 MiB/108 MiB CGO alloc/total, 121.3 CGO/sec, 69.1/2.8 %(u/s)time, 0.0 %gc (0x), 2.4 MiB/2.4 MiB (r/w)net
I200605 16:28:40.659353 311116 server/status/runtime.go:499  [n1] runtime stats: 5.9 GiB RSS, 396 goroutines, 622 MiB/231 MiB/1010 MiB GO alloc/idle/total, 63 MiB/108 MiB CGO alloc/total, 0.2 CGO/sec, 64.6/1.6 %(u/s)time, 0.0 %gc (0x), 2.3 MiB/2.3 MiB (r/w)net
I200605 16:28:40.929596 311217 kv/kvserver/replica_rangefeed.go:610  [n1,s1,r6/1:/Table/{SystemCon…-11}] RangeFeed closed timestamp 1591374505.914164767,0 is behind by 15.014988343s
I200605 16:28:46.134712 309394 server/status/runtime.go:499  [n1] runtime stats: 5.9 GiB RSS, 395 goroutines, 651 MiB/206 MiB/1010 MiB GO alloc/idle/total, 79 MiB/124 MiB CGO alloc/total, 0.3 CGO/sec, 69.2/1.8 %(u/s)time, 0.0 %gc (0x), 2.2 MiB/2.2 MiB (r/w)net
I200605 16:28:46.242533 444821 jobs/registry.go:812  [n1] job 561329284262232065: stepping through state reverting with error: committing system systemTable restoration: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORT_SPAN): "system-restore-txn" meta={id=f26c8d4a key=/Table/SystemConfigSpan/Start pri=99.99999995 epo=0 ts=1591374508.478949942,0 min=1591374508.478949942,0 seq=3010} lock=true stat=ABORTED rts=1591374508.478949942,0 wto=false max=1591374508.478949942,0
I200605 16:28:46.554260 444821 jobs/registry.go:812  [n1] job 561329284262232065: stepping through state failed with error: job 561329284262232065: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"@4>A\003\n\251\003\n\005users\030: 9(\003:\n\010\274\301\357\325\232\314\354\212\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\274\301\357\325\232\314\354\212\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591374508227731527 > 
E200605 16:28:46.708522 444821 jobs/registry.go:962  [n1] job 561329284262232065: adoption completed with error job 561329284262232065: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"@4>A\003\n\251\003\n\005users\030: 9(\003:\n\010\274\301\357\325\232\314\354\212\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\274\301\357\325\232\314\354\212\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591374508227731527 > 
I200605 16:28:46.746028 444821 jobs/registry.go:968  [n1] job 561329284262232065: status failed after adoption finished
I200605 16:28:46.752505 309120 util/stop/stopper.go:539  quiescing
I200605 16:28:46.753146 446053 util/stop/stopper.go:539  quiescing
I200605 16:28:46.936465 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200605 16:28:46.936814 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200605 16:28:46.936984 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (split): node unavailable; try another peer
I200605 16:28:46.938826 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200605 16:28:46.939713 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200605 16:28:46.940553 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200605 16:28:46.941464 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200605 16:28:46.942064 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200605 16:28:46.942721 311235 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200605 16:28:47.087233 311211 kv/kvserver/store.go:1652  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 16:28:47.124272 309120 util/stop/stopper.go:539  quiescing
I200605 16:28:47.134093 446036 util/stop/stopper.go:539  quiescing
I200605 16:28:47.144411 309391 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 415 iterations on up to 33814 requests
I200605 16:28:47.272445 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200605 16:28:47.272765 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200605 16:28:47.272905 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (split): node unavailable; try another peer
I200605 16:28:47.273058 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200605 16:28:47.273217 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200605 16:28:47.275820 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200605 16:28:47.275992 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200605 16:28:47.276117 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200605 16:28:47.276229 309387 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
W200605 16:28:47.472830 309395 ts/db.go:194  [n1,ts-poll] error writing time series data: node unavailable; try another peer
--- FAIL: TestFullClusterBackup (351.76s)
    full_cluster_backup_restore_test.go:136: error executing 'RESTORE FROM $1': pq: job 561329284262232065: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"@4>A\003\n\251\003\n\005users\030: 9(\003:\n\010\274\301\357\325\232\314\354\212\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\274\301\357\325\232\314\354\212\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591374508227731527 > 

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestFullClusterBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestFullClusterBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestFullClusterBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1996248&tab=buildLog) on [master@b86ced274f4458192c0e2f33ede84e7456861b01](https://github.com/cockroachdb/cockroach/commits/b86ced274f4458192c0e2f33ede84e7456861b01):

```
I200608 06:06:38.650337 443811 kv/kvserver/replica_command.go:413  [n1,s1,r92/1:/Table/6{2-4}] initiating a split of this range at key /Table/62/2 [r97] (manual)
I200608 06:06:38.860491 443812 kv/kvserver/replica_command.go:413  [n1,s1,r96/1:/{Table/64/2-Max}] initiating a split of this range at key /Table/64/3 [r98] (manual)
I200608 06:06:38.949665 308847 server/status/runtime.go:499  [n1] runtime stats: 5.9 GiB RSS, 411 goroutines, 731 MiB/184 MiB/1002 MiB GO alloc/idle/total, 65 MiB/115 MiB CGO alloc/total, 165.0 CGO/sec, 104.7/9.2 %(u/s)time, 0.0 %gc (0x), 2.3 MiB/2.3 MiB (r/w)net
I200608 06:06:38.960696 443811 kv/kvserver/replica_command.go:413  [n1,s1,r97/1:/Table/6{2/2-4}] initiating a split of this range at key /Table/62/3 [r99] (manual)
I200608 06:06:39.107946 443812 kv/kvserver/replica_command.go:413  [n1,s1,r98/1:/{Table/64/3-Max}] initiating a split of this range at key /Table/65 [r100] (manual)
I200608 06:06:39.234127 443811 kv/kvserver/replica_command.go:413  [n1,s1,r99/1:/Table/6{2/3-4}] initiating a split of this range at key /Table/63 [r101] (manual)
I200608 06:06:39.423568 443811 kv/kvserver/replica_command.go:413  [n1,s1,r95/1:/Table/5{3-8}] initiating a split of this range at key /Table/53/1/1 [r102] (manual)
I200608 06:06:39.430104 443810 kv/kvserver/replica_command.go:413  [n1,s1,r95/1:/Table/5{3-8}] initiating a split of this range at key /Table/53/1/4 [r103] (manual)
I200608 06:06:39.629908 443811 kv/kvserver/replica_command.go:413  [n1,s1,r102/1:/Table/5{3/1/1-8}] initiating a split of this range at key /Table/53/1/2 [r104] (manual)
I200608 06:06:39.741346 443810 kv/kvserver/replica_command.go:413  [n1,s1,r102/1:/Table/5{3/1/1-8}] initiating a split of this range at key /Table/53/1/4 [r105] (manual)
I200608 06:06:39.849071 443811 kv/kvserver/replica_command.go:413  [n1,s1,r104/1:/Table/5{3/1/2-8}] initiating a split of this range at key /Table/53/1/3 [r106] (manual)
I200608 06:06:39.879932 443810 kv/kvserver/replica_command.go:413  [n1,s1,r104/1:/Table/5{3/1/2-8}] initiating a split of this range at key /Table/53/1/4 [r107] (manual)
I200608 06:06:40.101622 443810 kv/kvserver/replica_command.go:413  [n1,s1,r106/1:/Table/5{3/1/3-8}] initiating a split of this range at key /Table/53/1/4 [r108] (manual)
I200608 06:06:40.306172 443812 kv/kvserver/replica_command.go:413  [n1,s1,r108/1:/Table/5{3/1/4-8}] initiating a split of this range at key /Table/53/1/5 [r109] (manual)
I200608 06:06:40.318622 443810 kv/kvserver/replica_command.go:413  [n1,s1,r108/1:/Table/5{3/1/4-8}] initiating a split of this range at key /Table/53/1/8 [r110] (manual)
I200608 06:06:40.486837 443812 kv/kvserver/replica_command.go:413  [n1,s1,r109/1:/Table/5{3/1/5-8}] initiating a split of this range at key /Table/53/1/6 [r111] (manual)
I200608 06:06:40.526201 443810 kv/kvserver/replica_command.go:413  [n1,s1,r109/1:/Table/5{3/1/5-8}] initiating a split of this range at key /Table/53/1/8 [r112] (manual)
I200608 06:06:40.781298 443812 kv/kvserver/replica_command.go:413  [n1,s1,r111/1:/Table/5{3/1/6-8}] initiating a split of this range at key /Table/53/1/7 [r113] (manual)
I200608 06:06:40.811233 443810 kv/kvserver/replica_command.go:413  [n1,s1,r111/1:/Table/5{3/1/6-8}] initiating a split of this range at key /Table/53/1/8 [r114] (manual)
I200608 06:06:41.016096 443810 kv/kvserver/replica_command.go:413  [n1,s1,r113/1:/Table/5{3/1/7-8}] initiating a split of this range at key /Table/53/1/8 [r115] (manual)
I200608 06:06:41.191697 443811 kv/kvserver/replica_command.go:413  [n1,s1,r115/1:/Table/5{3/1/8-8}] initiating a split of this range at key /Table/53/1/9 [r116] (manual)
I200608 06:06:41.469102 443811 kv/kvserver/replica_command.go:413  [n1,s1,r116/1:/Table/5{3/1/9-8}] initiating a split of this range at key /Table/54 [r117] (manual)
I200608 06:06:41.655382 443811 kv/kvserver/replica_command.go:413  [n1,s1,r117/1:/Table/5{4-8}] initiating a split of this range at key /Table/56 [r118] (manual)
I200608 06:06:42.726608 310790 server/status/runtime.go:499  [n1] runtime stats: 5.9 GiB RSS, 399 goroutines, 763 MiB/158 MiB/1002 MiB GO alloc/idle/total, 65 MiB/114 MiB CGO alloc/total, 501.9 CGO/sec, 101.0/7.5 %(u/s)time, 0.0 %gc (0x), 2.5 MiB/2.5 MiB (r/w)net
I200608 06:06:48.950920 308847 server/status/runtime.go:499  [n1] runtime stats: 5.9 GiB RSS, 393 goroutines, 795 MiB/129 MiB/1002 MiB GO alloc/idle/total, 64 MiB/114 MiB CGO alloc/total, 337.7 CGO/sec, 84.9/3.8 %(u/s)time, 0.0 %gc (0x), 2.7 MiB/2.7 MiB (r/w)net
I200608 06:06:52.737564 310790 server/status/runtime.go:499  [n1] runtime stats: 5.9 GiB RSS, 395 goroutines, 809 MiB/116 MiB/1002 MiB GO alloc/idle/total, 64 MiB/114 MiB CGO alloc/total, 0.2 CGO/sec, 70.1/1.9 %(u/s)time, 0.0 %gc (0x), 2.7 MiB/2.7 MiB (r/w)net
I200608 06:06:55.235882 310766 kv/kvserver/replica_rangefeed.go:610  [n1,s1,r6/1:/Table/{SystemCon…-11}] RangeFeed closed timestamp 1591596400.223575520,0 is behind by 15.01191618s
I200608 06:06:58.952448 308847 server/status/runtime.go:499  [n1] runtime stats: 5.9 GiB RSS, 397 goroutines, 848 MiB/78 MiB/1002 MiB GO alloc/idle/total, 80 MiB/130 MiB CGO alloc/total, 0.3 CGO/sec, 69.6/2.5 %(u/s)time, 0.0 %gc (0x), 2.4 MiB/2.4 MiB (r/w)net
I200608 06:06:59.102402 443709 jobs/registry.go:812  [n1] job 562056385069907969: stepping through state reverting with error: committing system systemTable restoration: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_ABORTED_RECORD_FOUND): meta={id=4bbcfe00 key=/Table/SystemConfigSpan/Start pri=99.99999995 epo=0 ts=1591596412.589121224,0 min=1591596402.556631901,0 seq=0} lock=true stat=ABORTED rts=0,0 wto=false max=0,0 int=1
I200608 06:06:59.432425 443709 jobs/registry.go:812  [n1] job 562056385069907969: stepping through state failed with error: job 562056385069907969: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"F<a\006\003\n\251\003\n\005users\030: 9(\003:\n\010\270\345\357\230\267\206\237\213\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\270\345\357\230\267\206\237\213\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591596402304240632 > 
E200608 06:06:59.489963 443709 jobs/registry.go:962  [n1] job 562056385069907969: adoption completed with error job 562056385069907969: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"F<a\006\003\n\251\003\n\005users\030: 9(\003:\n\010\270\345\357\230\267\206\237\213\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\270\345\357\230\267\206\237\213\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591596402304240632 > 
I200608 06:06:59.522598 443709 jobs/registry.go:968  [n1] job 562056385069907969: status failed after adoption finished
I200608 06:06:59.545987 308790 util/stop/stopper.go:539  quiescing
I200608 06:06:59.546565 444753 util/stop/stopper.go:539  quiescing
W200608 06:06:59.559912 444750 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
I200608 06:06:59.701223 308790 util/stop/stopper.go:539  quiescing
I200608 06:06:59.705273 444828 util/stop/stopper.go:539  quiescing
W200608 06:06:59.769809 308848 ts/db.go:197  [n1,ts-poll] node unavailable; try another peer
I200608 06:07:00.188675 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200608 06:07:00.189191 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200608 06:07:00.189349 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (split): node unavailable; try another peer
I200608 06:07:00.189557 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200608 06:07:00.189759 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200608 06:07:00.189975 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200608 06:07:00.190219 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200608 06:07:00.190377 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200608 06:07:00.190526 308972 kv/kvserver/queue.go:580  [n1,s1] rate limited in MaybeAdd (timeSeriesMaintenance): node unavailable; try another peer
I200608 06:07:00.681260 308994 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 301 iterations on up to 33964 requests
--- FAIL: TestFullClusterBackup (342.25s)
    full_cluster_backup_restore_test.go:136: error executing 'RESTORE FROM $1': pq: job 562056385069907969: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"F<a\006\003\n\251\003\n\005users\030: 9(\003:\n\010\270\345\357\230\267\206\237\213\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\270\345\357\230\267\206\237\213\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591596402304240632 > 

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestFullClusterBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestFullClusterBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestFullClusterBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2004584&tab=buildLog) on [master@00f54db3961ed7915e10d0370c8244446d46b97e](https://github.com/cockroachdb/cockroach/commits/00f54db3961ed7915e10d0370c8244446d46b97e):

```
I200611 03:55:34.934095 444366 kv/kvserver/replica_command.go:413  [n1,s1,r96/1:/{Table/64/2-Max}] initiating a split of this range at key /Table/64/3 [r98] (manual)
I200611 03:55:35.200907 444367 kv/kvserver/replica_command.go:413  [n1,s1,r95/1:/Table/5{3-8}] initiating a split of this range at key /Table/53/1/1 [r99] (manual)
I200611 03:55:35.201942 444365 kv/kvserver/replica_command.go:413  [n1,s1,r95/1:/Table/5{3-8}] initiating a split of this range at key /Table/53/1/4 [r100] (manual)
I200611 03:55:35.336682 444366 kv/kvserver/replica_command.go:413  [n1,s1,r98/1:/{Table/64/3-Max}] initiating a split of this range at key /Table/65 [r101] (manual)
I200611 03:55:35.531410 444366 kv/kvserver/replica_command.go:413  [n1,s1,r100/1:/Table/5{3/1/4-8}] initiating a split of this range at key /Table/53/1/5 [r102] (manual)
I200611 03:55:35.534244 444365 kv/kvserver/replica_command.go:413  [n1,s1,r100/1:/Table/5{3/1/4-8}] initiating a split of this range at key /Table/53/1/8 [r103] (manual)
I200611 03:55:35.541332 444367 kv/kvserver/replica_command.go:413  [n1,s1,r95/1:/Table/53{-/1/4}] initiating a split of this range at key /Table/53/1/1 [r104] (manual)
I200611 03:55:35.777823 444367 kv/kvserver/replica_command.go:413  [n1,s1,r104/1:/Table/53/1/{1-4}] initiating a split of this range at key /Table/53/1/2 [r105] (manual)
I200611 03:55:35.804012 444366 kv/kvserver/replica_command.go:413  [n1,s1,r102/1:/Table/5{3/1/5-8}] initiating a split of this range at key /Table/53/1/6 [r106] (manual)
I200611 03:55:35.861938 444365 kv/kvserver/replica_command.go:413  [n1,s1,r102/1:/Table/5{3/1/5-8}] initiating a split of this range at key /Table/53/1/8 [r107] (manual)
I200611 03:55:35.999592 444367 kv/kvserver/replica_command.go:413  [n1,s1,r105/1:/Table/53/1/{2-4}] initiating a split of this range at key /Table/53/1/3 [r108] (manual)
I200611 03:55:36.065258 444366 kv/kvserver/replica_command.go:413  [n1,s1,r106/1:/Table/5{3/1/6-8}] initiating a split of this range at key /Table/53/1/7 [r109] (manual)
I200611 03:55:36.153574 310506 server/status/runtime.go:504  [n1] runtime stats: 7.6 GiB RSS, 445 goroutines, 794 MiB/395 MiB/1.3 GiB GO alloc/idle/total, 68 MiB/109 MiB CGO alloc/total, 399.4 CGO/sec, 108.9/8.7 %(u/s)time, 0.0 %gc (0x), 1.9 MiB/1.9 MiB (r/w)net
I200611 03:55:36.156032 444365 kv/kvserver/replica_command.go:413  [n1,s1,r106/1:/Table/5{3/1/6-8}] initiating a split of this range at key /Table/53/1/8 [r110] (manual)
I200611 03:55:36.366332 444365 kv/kvserver/replica_command.go:413  [n1,s1,r109/1:/Table/5{3/1/7-8}] initiating a split of this range at key /Table/53/1/8 [r111] (manual)
I200611 03:55:36.527348 444367 kv/kvserver/replica_command.go:413  [n1,s1,r111/1:/Table/5{3/1/8-8}] initiating a split of this range at key /Table/53/1/9 [r112] (manual)
I200611 03:55:36.728881 444367 kv/kvserver/replica_command.go:413  [n1,s1,r112/1:/Table/5{3/1/9-8}] initiating a split of this range at key /Table/54 [r113] (manual)
I200611 03:55:36.894457 444367 kv/kvserver/replica_command.go:413  [n1,s1,r113/1:/Table/5{4-8}] initiating a split of this range at key /Table/56 [r114] (manual)
I200611 03:55:42.104644 308769 server/status/runtime.go:504  [n1] runtime stats: 7.6 GiB RSS, 427 goroutines, 825 MiB/373 MiB/1.3 GiB GO alloc/idle/total, 67 MiB/109 MiB CGO alloc/total, 498.9 CGO/sec, 93.1/5.9 %(u/s)time, 0.0 %gc (0x), 2.2 MiB/2.2 MiB (r/w)net
I200611 03:55:46.171723 310506 server/status/runtime.go:504  [n1] runtime stats: 7.5 GiB RSS, 427 goroutines, 840 MiB/361 MiB/1.3 GiB GO alloc/idle/total, 66 MiB/109 MiB CGO alloc/total, 103.6 CGO/sec, 68.3/2.4 %(u/s)time, 0.0 %gc (0x), 2.3 MiB/2.3 MiB (r/w)net
I200611 03:55:50.391752 310614 kv/kvserver/replica_rangefeed.go:610  [n1,s1,r6/1:/Table/{SystemCon…-11}] RangeFeed closed timestamp 1591847735.375664106,0 is behind by 15.016047478s
I200611 03:55:51.932612 308765 gossip/gossip.go:578  [n1] gossip status (ok, 1 node)
gossip client (0/3 cur/max conns)
gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I200611 03:55:52.044860 308769 server/status/runtime.go:504  [n1] runtime stats: 7.6 GiB RSS, 428 goroutines, 870 MiB/334 MiB/1.3 GiB GO alloc/idle/total, 66 MiB/109 MiB CGO alloc/total, 0.2 CGO/sec, 65.9/2.3 %(u/s)time, 0.0 %gc (0x), 2.1 MiB/2.1 MiB (r/w)net
I200611 03:55:55.804919 310614 kv/kvserver/replica_rangefeed.go:610  [n1,s1,r6/1:/Table/{SystemCon…-11}] RangeFeed closed timestamp 1591847735.375664106,0 is behind by 20.429218678s
I200611 03:55:56.092704 310502 gossip/gossip.go:578  [n1] gossip status (ok, 1 node)
gossip client (0/3 cur/max conns)
gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I200611 03:55:56.156067 310506 server/status/runtime.go:504  [n1] runtime stats: 7.6 GiB RSS, 427 goroutines, 888 MiB/319 MiB/1.3 GiB GO alloc/idle/total, 66 MiB/109 MiB CGO alloc/total, 0.2 CGO/sec, 67.1/2.6 %(u/s)time, 0.0 %gc (0x), 2.1 MiB/2.1 MiB (r/w)net
W200611 03:55:57.413603 310594 kv/kvserver/store_raft.go:509  [n1,s1,r6/1:/Table/{SystemCon…-11}] handle raft ready: 0.6s [applied=1, batches=1, state_assertions=0]
I200611 03:55:57.522482 444360 jobs/registry.go:813  [n1] job 562879963902738433: stepping through state reverting with error: committing system systemTable restoration: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "system-restore-txn" meta={id=721176fb key=/Table/SystemConfigSpan/Start pri=0.02791099 epo=0 ts=1591847737.895999803,0 min=1591847737.895999803,0 seq=3010} lock=true stat=PENDING rts=1591847737.895999803,0 wto=false max=1591847738.395999803,0
I200611 03:55:57.654839 444360 jobs/registry.go:813  [n1] job 562879963902738433: stepping through state failed with error: job 562879963902738433: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"\315\340-_\003\n\251\003\n\005users\030: 9(\003:\n\010\251\360\325\377\314\230\330\213\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\251\360\325\377\314\230\330\213\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591847737718994159 > 
E200611 03:55:57.860745 444360 jobs/registry.go:963  [n1] job 562879963902738433: adoption completed with error job 562879963902738433: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"\315\340-_\003\n\251\003\n\005users\030: 9(\003:\n\010\251\360\325\377\314\230\330\213\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\251\360\325\377\314\230\330\213\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591847737718994159 > 
I200611 03:55:57.890753 444360 jobs/registry.go:969  [n1] job 562879963902738433: status failed after adoption finished
I200611 03:55:57.897304 308585 util/stop/stopper.go:539  quiescing
I200611 03:55:57.898204 445246 util/stop/stopper.go:539  quiescing
W200611 03:55:57.899497 445570 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W200611 03:55:58.144355 310507 ts/db.go:194  [n1,ts-poll] error writing time series data: node unavailable; try another peer
W200611 03:55:58.170057 310644 jobs/registry.go:1246  job 562879813208309761: canceling due to liveness failure
W200611 03:55:58.170636 310644 jobs/registry.go:1246  job 562879829210398721: canceling due to liveness failure
W200611 03:55:58.170803 310644 jobs/registry.go:1246  job 562879845060935681: canceling due to liveness failure
W200611 03:55:58.171840 310644 jobs/registry.go:1246  job 562879873729953793: canceling due to liveness failure
W200611 03:55:58.172169 310644 jobs/registry.go:1246  job 562879922890145793: canceling due to liveness failure
W200611 03:55:58.172335 310644 jobs/registry.go:1246  job 562879716138844161: canceling due to liveness failure
I200611 03:55:58.238293 308585 util/stop/stopper.go:539  quiescing
I200611 03:55:58.239057 445603 util/stop/stopper.go:539  quiescing
I200611 03:55:58.421835 308619 kv/kvclient/kvcoord/transport_race.go:108  transport race promotion: ran 338 iterations on up to 33875 requests
--- FAIL: TestFullClusterBackup (366.96s)
    full_cluster_backup_restore_test.go:136: error executing 'RESTORE FROM $1': pq: job 562879963902738433: cannot be reverted, manual cleanup may be required: dropping tables caused by restore fail/cancel: unexpected value: raw_bytes:"\315\340-_\003\n\251\003\n\005users\030: 9(\003:\n\010\251\360\325\377\314\230\330\213\026B\"\n\010username\020\001\032\014\010\007\020\000\030\0000\000P\031`\000 \0000\000h\000p\000B(\n\016hashedPassword\020\002\032\014\010\010\020\000\030\0000\000P\021`\000 \0010\000h\000p\000B'\n\006isRole\020\003\032\014\010\000\020\000\030\0000\000P\020`\000 \000*\005false0\000h\000p\000H\004RQ\n\007primary\020\001\030\001\"\010username0\001@\000J\020\010\000\020\000\032\000 \000(\0000\0008\000@\000Z\000z\002\010\000\200\001\000\210\001\000\220\001\001\230\001\000\242\001\006\010\000\022\000\030\000\250\001\000\262\001\000\272\001\000`\002j\025\n\t\n\005admin\020\002\n\010\n\004root\020\002\200\001\001\210\001\003\230\001\000\262\001\031\n\007primary\020\000\032\010username \001(\000\262\001,\n\024fam_2_hashedPassword\020\002\032\016hashedPassword \002(\002\262\001\034\n\014fam_3_isRole\020\003\032\006isRole \003(\003\270\001\004\302\001\000\350\001\000\362\001\004\010\000\022\000\370\001\000\200\002\000\222\002\000\232\002\n\010\251\360\325\377\314\230\330\213\026\262\002\trestoring\270\002\000\300\002\035" timestamp:<wall_time:1591847737718994159 > 

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestFullClusterBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestFullClusterBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -1186,7 +1186,13 @@ func TestRestoreFailCleanup(t *testing.T) {
 	// Disable GC job so that the final check of crdb_internal.tables is
 	// guaranteed to not be cleaned up. Although this was never observed by a
 	// stress test, it is here for safety.
-	params.Knobs.GCJob = &sql.GCJobTestingKnobs{RunBeforeResume: func(_ int64) error { select {} /* blocks forever */ }}
+	blockGC := make(chan struct{})
+	params.Knobs.GCJob = &sql.GCJobTestingKnobs{
+		RunBeforeResume: func(_ int64) error {
+			<-blockGC
+			return nil
+		},
+	}
 
 	const numAccounts = 1000
 	_, _, sqlDB, dir, cleanup := backupRestoreTestSetupWithParams(t, singleNode, numAccounts,

--- a/pkg/ccl/backupccl/full_cluster_backup_restore_test.go
+++ b/pkg/ccl/backupccl/full_cluster_backup_restore_test.go
@@ -14,9 +14,14 @@ import (
 	"strconv"
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/base"
 	_ "github.com/cockroachdb/cockroach/pkg/ccl/partitionccl"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/errors"
 )
 
 // Large test to ensure that all of the system table data is being restored in
@@ -352,4 +357,70 @@ func TestCreateDBAndTableIncrementalFullClusterBackup(t *testing.T) {
 
 	// Ensure that the new backup succeeds.
 	sqlDB.Exec(t, `BACKUP TO $1`, LocalFoo)
+}
+
+// TestClusterRestoreFailCleanup tests that a failed RESTORE is cleaned up.
+func TestClusterRestoreFailCleanup(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	params := base.TestServerArgs{}
+	// Disable GC job so that the final check of crdb_internal.tables is
+	// guaranteed to not be cleaned up. Although this was never observed by a
+	// stress test, it is here for safety.
+	blockCh := make(chan struct{})
+	defer close(blockCh)
+	params.Knobs.GCJob = &sql.GCJobTestingKnobs{
+		RunBeforeResume: func(_ int64) error { <-blockCh; return nil },
+	}
+
+	const numAccounts = 1000
+	_, _, sqlDB, tempDir, cleanupFn := BackupRestoreTestSetup(t, singleNode, numAccounts, InitNone)
+	_, tcRestore, sqlDBRestore, cleanupEmptyCluster := backupRestoreTestSetupEmpty(
+		t, singleNode, tempDir, InitNone,
+	)
+	defer cleanupFn()
+	defer cleanupEmptyCluster()
+
+	// Setup the system systemTablesToVerify to ensure that they are copied to the new cluster.
+	// Populate system.users.
+	for i := 0; i < 1000; i++ {
+		sqlDB.Exec(t, fmt.Sprintf("CREATE USER maxroach%d", i))
+	}
+	sqlDB.Exec(t, `BACKUP TO $1`, LocalFoo)
+
+	// Bugger the backup by injecting a failure while restoring the system data.
+	for _, server := range tcRestore.Servers {
+		registry := server.JobRegistry().(*jobs.Registry)
+		registry.TestingResumerCreationKnobs = map[jobspb.Type]func(raw jobs.Resumer) jobs.Resumer{
+			jobspb.TypeRestore: func(raw jobs.Resumer) jobs.Resumer {
+				r := raw.(*restoreResumer)
+				r.testingKnobs.duringSystemTableRestoration = func() error {
+					return errors.New("injected error")
+				}
+				return r
+			},
+		}
+	}
+
+	sqlDBRestore.ExpectErr(
+		t, "injected error",
+		`RESTORE FROM $1`, LocalFoo,
+	)
+	// Verify the failed RESTORE added some DROP tables.
+	// Note that the system tables here correspond to the temporary tables
+	// imported, not the system tables themselves.
+	sqlDBRestore.CheckQueryResults(t,
+		`SELECT name FROM crdb_internal.tables WHERE state = 'DROP' ORDER BY name`,
+		[][]string{
+			{"bank"},
+			{"comments"},
+			{"jobs"},
+			{"locations"},
+			{"role_members"},
+			{"settings"},
+			{"ui"},
+			{"users"},
+			{"zones"},
+		},
+	)
 }

--- a/pkg/ccl/backupccl/restore_job.go
+++ b/pkg/ccl/backupccl/restore_job.go
@@ -795,6 +795,13 @@ type restoreResumer struct {
 	descriptorCoverage tree.DescriptorCoverage
 	latestStats        []*stats.TableStatisticProto
 	execCfg            *sql.ExecutorConfig
+
+	testingKnobs struct {
+		// duringSystemTableResotration is called once for every system table we
+		// restore. It is used to simulate any errors that we may face at this point
+		// of the restore.
+		duringSystemTableRestoration func() error
+	}
 }
 
 // remapRelevantStatistics changes the table ID references in the stats
@@ -1011,6 +1018,9 @@ func (r *restoreResumer) Resume(
 		return err
 	}
 
+	// TODO(pbardea): This was part of the original design where full cluster
+	// restores were a special case, but really we should be making only the
+	// temporary system tables public before we restore all the system table data.
 	if r.descriptorCoverage == tree.AllDescriptors {
 		if err := r.restoreSystemTables(ctx, p.ExecCfg().DB); err != nil {
 			return err
@@ -1085,6 +1095,7 @@ func (r *restoreResumer) publishTables(ctx context.Context) error {
 		// Write the new TableDescriptors and flip state over to public so they can be
 		// accessed.
 		b := txn.NewBatch()
+		newTables := make([]*sqlbase.TableDescriptor, 0, len(details.TableDescs))
 		for _, tbl := range r.tables {
 			newTableDesc := sqlbase.NewMutableExistingTableDescriptor(*tbl.TableDesc())
 			newTableDesc.Version++
@@ -1105,6 +1116,7 @@ func (r *restoreResumer) publishTables(ctx context.Context) error {
 				newTableDesc.DescriptorProto(),
 				existingDescVal,
 			)
+			newTables = append(newTables, newTableDesc.TableDesc())
 		}
 
 		if err := txn.Run(ctx, b); err != nil {
@@ -1113,6 +1125,7 @@ func (r *restoreResumer) publishTables(ctx context.Context) error {
 
 		// Update and persist the state of the job.
 		details.TablesPublished = true
+		details.TableDescs = newTables
 		if err := r.job.WithTxn(txn).SetDetails(ctx, details); err != nil {
 			for _, newJob := range newSchemaChangeJobs {
 				if cleanupErr := newJob.CleanupOnRollback(ctx); cleanupErr != nil {
@@ -1275,6 +1288,12 @@ func (r *restoreResumer) restoreSystemTables(ctx context.Context, db *kv.DB) err
 			return nil
 		}); err != nil {
 			return err
+		}
+
+		if fn := r.testingKnobs.duringSystemTableRestoration; fn != nil {
+			if err := fn(); err != nil {
+				return err
+			}
 		}
 	}
 

